plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -359,10 +383,12 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         if (TIME_WITH_TIME_ZONE.equals(type) || TIMESTAMP_TZ_MILLIS.equals(type)) {
             throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type.getDisplayName());
         }
-        if (TIMESTAMP_MILLIS.equals(type)) {
-            // TODO use `timestampWriteFunction` (https://github.com/trinodb/trino/issues/6910)
-            return WriteMapping.longMapping("datetime(3)", timestampWriteFunctionUsingSqlTimestamp(TIMESTAMP_MILLIS));
+        if (type instanceof TimestampType) {
+            TimestampType timestampType = (TimestampType) type;
+            verify(timestampType.getPrecision() <= MAX_SUPPORTED_TIMESTAMP_PRECISION);
```

Comment:
We can perform rounding when writing higher precisions similar to how we do it in Postgres connector.
Failure doesn't seem like a good-behaviour here IMO.

WDYT @findepi ?

Reply:
agreed. PostgreSQL behavior was deliberately picked and i don't see a reason to deviate from it here.

Reply:
@guyco33 Was this addressed? I don't see any explicit rounding performed for higher precision values. So it looks like we depend on driver behaviour?

For reference this is how Postgres does it https://github.com/trinodb/trino/blob/1ffe39138a7fccfc0a26a4288df7f9df3e2ae28d/plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java#L860-L873.